### PR TITLE
Stream worker REST API Example update

### PR DESCRIPTION
**Pull request by diana-macrometa**
- Moved and renamed file
- Tested and updated file

### CODE REVIEW

**Comment by diana-macrometa** [comment]
@koshy-thomas, I am having some issues with the JavaScript. I get this when I try to run it. I see errors, but it looks like I am also getting results? I do not know what is going on.

```
(base) dianapayton@Dianas-MacBook-Pro js % node sw-api-example.js
(node:74822) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
Stream worker already exists, skipping creation
Response {
  [Symbol(realm)]: null,
  [Symbol(state)]: {
    aborted: false,
    rangeRequested: false,
    timingAllowPassed: true,
    requestIncludesCredentials: true,
    type: 'default',
    status: 500,
    timingInfo: {
      startTime: 312.4398329257965,
      redirectStartTime: 0,
      redirectEndTime: 0,
      postRedirectStartTime: 312.4398329257965,
      finalServiceWorkerStartTime: 0,
      finalNetworkResponseStartTime: 0,
      finalNetworkRequestStartTime: 0,
      endTime: 0,
      encodedBodySize: 168,
      decodedBodySize: 168,
      finalConnectionTimingInfo: null
    },
    cacheState: '',
    statusText: 'Internal Server Error',
    headersList: HeadersList {
      [Symbol(headers map)]: [Map],
      [Symbol(headers map sorted)]: null
    },
    urlList: [ [URL] ],
    body: { stream: undefined }
  },
  [Symbol(headers)]: HeadersList {
    [Symbol(headers map)]: Map(17) {
      'content-type' => 'application/json',
      'content-length' => '168',
      'connection' => 'keep-alive',
      'date' => 'Wed, 04 Jan 2023 20:43:35 GMT',
      'x-xss-protection' => '1; mode=block',
      'x-frame-options' => 'sameorigin',
      'x-content-type-options' => 'nosniff',
      'access-control-allow-origin' => '*',
      'access-control-allow-headers' => 'origin, content-type, accept, authorization, x-applicationurl, x-requested-with, x-c8-frontend, x-c8-version',
      'access-control-expose-headers' => 'authorization, x-auth-token, Content-Disposition, etag, content-encoding, content-length, location, server, x-c8-errors, x-c8-async-id, x-gdn-region, x-gdn-requestid, x-gdn-responsetime',
      'access-control-allow-methods' => 'GET,POST,PUT,DELETE,HEAD,PATCH',
      'access-control-allow-credentials' => 'true',
      'server' => 'APISIX',
      'x-apisix-upstream-status' => '500',
      'x-gdn-requestid' => '0682f5f6-9da9-405e-ada9-49fb1c9a795e',
      'x-gdn-region' => 'play-us-west.paas.macrometa.io',
      'x-gdn-responsetime' => '24'
    },
    [Symbol(headers map sorted)]: null
  }
```

**Comment by diana-macrometa** [comment]
@koshy-thomas and @dlozina-macrometa , I'm getting 400 error with the Python example. Maybe we can chat tomorrow?

```
FED URL: https://api-play.paas.macrometa.io

stream worker Created:  {'error': True, 'code': 400, 'errorMessage': 'Response: 400, Error: 10 - bad parameter', 'errorNum': 400}

stream worker Activated:  {'error': True, 'code': 400, 'errorMessage': 'Response: 400, Error: 10 - bad parameter', 'errorNum': 400}
Traceback (most recent call last):
  File "/Users/dianapayton/Documents/python/sw-api-example.py", line 84, in <module>
    ws = create_connection(producerurl,header={"content-type": 'application/json', 'authorization': AUTH_TOKEN})
  File "/Users/dianapayton/opt/miniconda3/lib/python3.9/site-packages/websocket/_core.py", line 515, in create_connection
    websock.connect(url, **options)
  File "/Users/dianapayton/opt/miniconda3/lib/python3.9/site-packages/websocket/_core.py", line 226, in connect
    self.handshake_response = handshake(self.sock, *addrs, **options)
  File "/Users/dianapayton/opt/miniconda3/lib/python3.9/site-packages/websocket/_handshake.py", line 80, in handshake
    status, resp = _get_resp_headers(sock)
  File "/Users/dianapayton/opt/miniconda3/lib/python3.9/site-packages/websocket/_handshake.py", line 165, in _get_resp_headers
    raise WebSocketBadStatusException("Handshake status %d %s", status, status_message, resp_headers)
websocket._exceptions.WebSocketBadStatusException: Handshake status 401 Unauthorized
```

File: docs/streams/stream-examples/rest-apis-stream-example.md [hunk 1]
```diff
@@ -245,9 +243,9 @@ import time
 
 # Constants
 URL = "api-play.paas.macrometa.io"
-HTTP_URL = f"https://{URL}"
+HTTP_URL = "https://{URL}"
 API_KEY = "XXXXX" # Use your API key here
-AUTH_TOKEN = f"apikey {API_KEY}"
+AUTH_TOKEN = "apikey {API_KEY}"
```
Comment (koshy-thomas) [inline docs/streams/stream-examples/rest-apis-stream-example.md:248]: the f will be needed here

File: docs/streams/stream-examples/rest-apis-stream-example.md [hunk 2]
```diff
@@ -245,9 +243,9 @@ import time
 
 # Constants
 URL = "api-play.paas.macrometa.io"
-HTTP_URL = f"https://{URL}"
+HTTP_URL = "https://{URL}"
```
Comment (koshy-thomas) [inline docs/streams/stream-examples/rest-apis-stream-example.md:246]: the f will be needed here

File: docs/cep/examples/rest-apis-cep-example.md [hunk 3]
```diff
@@ -257,71 +109,65 @@ const streamAppDef = `@App:name('stream_app_tutorial')
...
-      ? federationName
+      ? url
       : `api-${dcUrl}`;
 
     const otpProducer = await connection.req(`/apid/otp`, {
       method: "POST"
     });
 
-    const producerUrl = `wss://${url}/_ws/ws/v2/producer/persistent/${tenant}/${region}._system/${streamName}?otp=${otpProducer.otp}`;
+    const producerUrl = `wss://${url}/_ws/ws/v2/producer/persistent/${url}/${region}._system/${streamName}?otp=${otpProducer.otp}`;
```
Comment (koshy-thomas) [inline docs/cep/examples/rest-apis-cep-example.md]:
```suggestion
    const producerUrl = `wss://${url}/_ws/ws/v2/producer/persistent/${tenant}/${region}._system/${streamName}?otp=${otpProducer.otp}`;
```

Comment (koshy-thomas) [inline docs/cep/examples/rest-apis-cep-example.md]:
```suggestion
    const producerUrl = `wss://${url}/_ws/ws/v2/producer/persistent/${tenant}/${region}._system/${streamName}?otp=${otpProducer.otp}`;
```

File: docs/cep/examples/rest-apis-cep-example.md [hunk 4]
```diff
@@ -388,21 +234,159 @@ const run = async function () {
...
+import requests
+import json
+import base64
+import six
+import time
+from websocket import create_connection
+
+# Constants  
+URL = "api-play.paas.macrometa.io"
+HTTP_URL = "https://{URL}"
```
Comment (koshy-thomas) [inline docs/cep/examples/rest-apis-cep-example.md]:
```suggestion
HTTP_URL = f"https://{URL}"
```

File: docs/cep/examples/rest-apis-cep-example.md [hunk 5]
```diff
@@ -388,21 +234,159 @@ const run = async function () {
...
+API_KEY = "XXXXX" # Use your API key here
+AUTH_TOKEN = "apikey {API_KEY}"
```
Comment (koshy-thomas) [inline docs/cep/examples/rest-apis-cep-example.md]:
```suggestion
AUTH_TOKEN = f"apikey {API_KEY}"
```

File: docs/streams/stream-examples/rest-apis-stream-example.md (hunk 2 above)
Comment (koshy-thomas) [inline docs/streams/stream-examples/rest-apis-stream-example.md]: f is needed here
```suggestion
HTTP_URL = f"https://{URL}"
```

File: docs/streams/stream-examples/rest-apis-stream-example.md (hunk 1 above)
Comment (koshy-thomas) [inline docs/streams/stream-examples/rest-apis-stream-example.md]: f is needed here
```suggestion
AUTH_TOKEN = f"apikey {API_KEY}"
```

File: docs/streams/stream-examples/rest-apis-stream-example.md (hunk 2 above)
Comment (koshy-thomas) [reply docs/streams/stream-examples/rest-apis-stream-example.md]: More about f-strings can be read here : https://www.geeksforgeeks.org/formatted-string-literals-f-strings-python/

**Comment by koshy-thomas** [comment]
@diana-macrometa it seems while moving or editing, some minor mistakes were made which broke the code like for example "f" was removed from some f-string variables in python sample. And in Javascript sample the websocket connection url requires the tenant name and it seems mistakenly you replaced the tenant with "url". Also since we are not using login anymore (JWT token) (we are using apikey now) so I removed it from the javascript sample here

**Comment by koshy-thomas** [comment]
@diana-macrometa please try all the samples in the PR now and let me know if it still doesn't work for you

**Comment by diana-macrometa** [comment]
I removed the f because my Python IDE was complaining that the syntax was wrong. I'll test again now.

**Comment by diana-macrometa** [comment]
@koshy-thomas - JavaScript is better, I think? There are a lot of nulls and an internal server error in there, so I am not sure. Also, I am not seeing the messages or that the stream worker was deleted.

```
(base) dianapayton@Dianas-MacBook-Pro js % node sw-api-example.js 
(node:92098) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
Stream worker already exists, skipping creation
Response {
  [Symbol(realm)]: null,
  [Symbol(state)]: {
    aborted: false,
    rangeRequested: false,
    timingAllowPassed: true,
    requestIncludesCredentials: true,
    type: 'default',
    status: 500,
    timingInfo: {
      startTime: 336.9302079677582,
      redirectStartTime: 0,
      redirectEndTime: 0,
      postRedirectStartTime: 336.9302079677582,
      finalServiceWorkerStartTime: 0,
      finalNetworkResponseStartTime: 0,
      finalNetworkRequestStartTime: 0,
      endTime: 0,
      encodedBodySize: 168,
      decodedBodySize: 168,
      finalConnectionTimingInfo: null
    },
    cacheState: '',
    statusText: 'Internal Server Error',
    headersList: HeadersList {
      [Symbol(headers map)]: [Map],
      [Symbol(headers map sorted)]: null
    },
    urlList: [ [URL] ],
    body: { stream: undefined }
  },
  [Symbol(headers)]: HeadersList {
    [Symbol(headers map)]: Map(17) {
      'content-type' => 'application/json',
      'content-length' => '168',
      'connection' => 'keep-alive',
      'date' => 'Thu, 05 Jan 2023 15:04:24 GMT',
      'x-xss-protection' => '1; mode=block',
      'x-frame-options' => 'sameorigin',
      'x-content-type-options' => 'nosniff',
      'access-control-allow-origin' => '*',
      'access-control-allow-headers' => 'origin, content-type, accept, authorization, x-applicationurl, x-requested-with, x-c8-frontend, x-c8-version',
      'access-control-expose-headers' => 'authorization, x-auth-token, Content-Disposition, etag, content-encoding, content-length, location, server, x-c8-errors, x-c8-async-id, x-gdn-region, x-gdn-requestid, x-gdn-responsetime',
      'access-control-allow-methods' => 'GET,POST,PUT,DELETE,HEAD,PATCH',
      'access-control-allow-credentials' => 'true',
      'server' => 'APISIX',
      'x-apisix-upstream-status' => '500',
      'x-gdn-requestid' => '14bc7d52-94ca-45ee-bded-a7edc7a00aa5',
      'x-gdn-region' => 'play-us-west.paas.macrometa.io',
      'x-gdn-responsetime' => '24'
    },
    [Symbol(headers map sorted)]: null
  }
}
```

**Comment by diana-macrometa** [comment]
@koshy-thomas, here is the Python output. At least I am getting in, but it is wildly different than the JS output. Also, I had to manually delete the stream workers in order to get it to work. Here is the output after I delete the stream workers:

```
(base) dianapayton@Dianas-MacBook-Pro python % python sw-api-example.py 
FED URL: https://api-play.paas.macrometa.io

stream worker Created:  {'error': False, 'code': 201, 'streamApps': [{'name': 'stream_worker_tutorial', 'definition': '\n  @App:name(\'stream_worker_tutorial\')\n  @App:qlVersion("2")\n  CREATE FUNCTION concatFn[javascript] return string {\n      var str1 = data[0];\n      var str2 = data[1];\n      var str3 = data[2];\n      var response = str1 + str2 + str3;\n      return response;\n  };\n  -- Stream\n  CREATE STREAM tutorialAppInputStream (deviceID string, roomNo int, temperature double);\n  -- Table\n  CREATE TABLE GLOBAL tutorialAppOutputTable (id string, temperature double);\n  @info(name=\'Query\')\n  INSERT INTO tutorialAppOutputTable\n  SELECT concatFn(roomNo,\'-\',deviceID) as id, temperature\n  FROM tutorialAppInputStream;\n', 'regions': ['play-us-west'], 'isActive': False}]}

stream worker Activated:  {'error': False, 'code': 200, 'streamApps': [{'name': 'stream_worker_tutorial', 'definition': '\n  @App:name(\'stream_worker_tutorial\')\n  @App:qlVersion("2")\n  CREATE FUNCTION concatFn[javascript] return string {\n      var str1 = data[0];\n      var str2 = data[1];\n      var str3 = data[2];\n      var response = str1 + str2 + str3;\n      return response;\n  };\n  -- Stream\n  CREATE STREAM tutorialAppInputStream (deviceID string, roomNo int, temperature double);\n  -- Table\n  CREATE TABLE GLOBAL tutorialAppOutputTable (id string, temperature double);\n  @info(name=\'Query\')\n  INSERT INTO tutorialAppOutputTable\n  SELECT concatFn(roomNo,\'-\',deviceID) as id, temperature\n  FROM tutorialAppInputStream;\n', 'regions': ['play-us-west'], 'isActive': True}]}
Message published successfully
Message published successfully

stream worker Results:  {'result': [{'_id': 'tutorialAppOutputTable/1983609465', '_key': '1983609465', '_rev': '_fWH5i2y--_', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1983609467', '_key': '1983609467', '_rev': '_fWH5i4a--_', 'id': '201-AD11', 'temperature': 47}], 'hasMore': False, 'cached': False, 'extra': {'stats': {'writesExecuted': 0, 'writesIgnored': 0, 'scannedFull': 2, 'scannedIndex': 0, 'filtered': 0, 'httpRequests': 0, 'executionTime': 0.00011491775512695312, 'peakMemoryUsage': 22640}, 'warnings': []}, 'error': False, 'code': 201}

stream worker Deleted:  {'error': False, 'code': 200, 'message': 'Stream worker `stream_worker_tutorial` deleted successfully.'}
```

**Comment by diana-macrometa** [comment]
Here is the output when the stream workers exist. So if the stream worker deletion isn't working, then we have some major issues with anyone who wants to run this more than once.

```
(base) dianapayton@Dianas-MacBook-Pro python % python sw-api-example.py
FED URL: https://api-play.paas.macrometa.io

stream worker Created:  {'error': True, 'code': 400, 'errorMessage': 'Response: 400, Error: 10 - bad parameter', 'errorNum': 400}

stream worker Activated:  {'error': True, 'code': 400, 'errorMessage': 'Response: 400, Error: 10 - bad parameter', 'errorNum': 400}
Traceback (most recent call last):
(base) dianapayton@Dianas-MacBook-Pro python % python sw-api-example.py 
FED URL: https://api-play.paas.macrometa.io

stream worker Created:  {'error': True, 'code': 409, 'errorMessage': "Stream application 'stream_worker_tutorial' already exists", 'errorNum': 409}
```

**Comment by diana-macrometa** [comment]
Latest JS output is looking better:

```
(base) dianapayton@Dianas-MacBook-Pro js % node sw-api-example.js
(node:96892) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
Stream worker created successfully {
  error: false,
  code: 201,
  streamApps: [
    {
      name: 'stream_worker_tutorial',
      definition: "@App:name('stream_worker_tutorial')\n" +
        "  @App:description('This application demonstrates how to use user-defined functions in the stream worker.')\n" +
        '  @App:qlVersion("2")\n' +
        '  CREATE FUNCTION concatFn[javascript] return string {\n' +
        '      var str1 = data[0];\n' +
        '      var str2 = data[1];\n' +
        '      var str3 = data[2];\n' +
        '      var response = str1 + str2 + str3;\n' +
        '      return response;\n' +
        '  };\n' +
        '  -- Stream\n' +
        '  CREATE STREAM tutorialAppInputStream (deviceID string, roomNo int, temperature double);\n' +
        '  -- Table\n' +
        '  CREATE TABLE GLOBAL tutorialAppOutputTable (id string, temperature double);\n' +
        "  @info(name='Query')\n" +
        '  INSERT INTO tutorialAppOutputTable\n' +
        "  SELECT concatFn(roomNo,'-',deviceID) as id, temperature\n" +
        '  FROM tutorialAppInputStream;',
      regions: [Array],
      isActive: false
    }
  ]
}
Activating stream worker... stream_worker_tutorial
Stream worker activated successfully
WebSocket:Producer is open now for c8locals.tutorialAppInputStream
WebSocket:Producer message sent successfully
WebSocket:Producer message sent successfully
Closed WebSocket:Producer connection for c8locals.tutorialAppInputStream
Input sent ---> [
  { deviceID: 'AD11', roomNo: 200, temperature: 18 },
  { deviceID: 'AD11', roomNo: 201, temperature: 47 }
]
Output received ---> [
  {
    _id: 'tutorialAppOutputTable/1983609465',
    _key: '1983609465',
    _rev: '_fWH5i2y--_',
    id: '200-AD11',
    temperature: 18
  },
  {
    _id: 'tutorialAppOutputTable/1983609467',
    _key: '1983609467',
    _rev: '_fWH5i4a--_',
    id: '201-AD11',
    temperature: 47
  },
  {
    _id: 'tutorialAppOutputTable/1984084894',
    _key: '1984084894',
    _rev: '_fWIMMmq--_',
    id: '200-AD11',
    temperature: 18
  },
  {
    _id: 'tutorialAppOutputTable/1984084896',
    _key: '1984084896',
    _rev: '_fWIMMmu--_',
    id: '201-AD11',
    temperature: 47
  },
  {
    _id: 'tutorialAppOutputTable/1984095390',
    _key: '1984095390',
    _rev: '_fWIMiHq--_',
    id: '200-AD11',
    temperature: 18
  },
  {
    _id: 'tutorialAppOutputTable/1984101014',
    _key: '1984101014',
    _rev: '_fWIMr5K--_',
    id: '201-AD11',
    temperature: 47
  },
  {
    _id: 'tutorialAppOutputTable/1984155228',
    _key: '1984155228',
    _rev: '_fWIO6Ze--_',
    id: '200-AD11',
    temperature: 18
  },
  {
    _id: 'tutorialAppOutputTable/1984155234',
    _key: '1984155234',
    _rev: '_fWIO6Zm--_',
    id: '201-AD11',
    temperature: 47
  },
  {
    _id: 'tutorialAppOutputTable/1984155236',
    _key: '1984155236',
    _rev: '_fWIO6Zm--C',
    id: '200-AD11',
    temperature: 18
  },
  {
    _id: 'tutorialAppOutputTable/1984155239',
    _key: '1984155239',
    _rev: '_fWIO6Zq--B',
    id: '201-AD11',
    temperature: 47
  },
  {
    _id: 'tutorialAppOutputTable/1984164734',
    _key: '1984164734',
    _rev: '_fWIPP0m--_',
    id: '200-AD11',
    temperature: 18
  },
  {
    _id: 'tutorialAppOutputTable/1984170401',
    _key: '1984170401',
    _rev: '_fWIPZk2--_',
    id: '201-AD11',
    temperature: 47
  }
]
Stream worker deleted successfully {
  error: false,
  code: 200,
  message: 'Stream worker `stream_worker_tutorial` deleted successfully.'
}
```

**Comment by diana-macrometa** [comment]
And here is the Python output:

```
(base) dianapayton@Dianas-MacBook-Pro python % python sw-api-example.py 
FED URL: https://api-play.paas.macrometa.io

stream worker Created:  {'error': False, 'code': 201, 'streamApps': [{'name': 'stream_worker_tutorial', 'definition': '\n  @App:name(\'stream_worker_tutorial\')\n  @App:qlVersion("2")\n  CREATE FUNCTION concatFn[javascript] return string {\n      var str1 = data[0];\n      var str2 = data[1];\n      var str3 = data[2];\n      var response = str1 + str2 + str3;\n      return response;\n  };\n  -- Stream\n  CREATE STREAM tutorialAppInputStream (deviceID string, roomNo int, temperature double);\n  -- Table\n  CREATE TABLE GLOBAL tutorialAppOutputTable (id string, temperature double);\n  @info(name=\'Query\')\n  INSERT INTO tutorialAppOutputTable\n  SELECT concatFn(roomNo,\'-\',deviceID) as id, temperature\n  FROM tutorialAppInputStream;\n', 'regions': ['play-us-west'], 'isActive': False}]}

stream worker Activated:  {'error': False, 'code': 200, 'streamApps': [{'name': 'stream_worker_tutorial', 'definition': '\n  @App:name(\'stream_worker_tutorial\')\n  @App:q
(base) dianapayton@Dianas-MacBook-Pro python % python sw-api-example.py 
FED URL: https://api-play.paas.macrometa.io

stream worker Created:  {'error': False, 'code': 201, 'streamApps': [{'name': 'stream_worker_tutorial', 'definition': '\n  @App:name(\'stream_worker_tutorial\')\n  @App:qlVersion("2")\n  CREATE FUNCTION concatFn[javascript] return string {\n      var str1 = data[0];\n      var str2 = data[1];\n      var str3 = data[2];\n      var response = str1 + str2 + str3;\n      return response;\n  };\n  -- Stream\n  CREATE STREAM tutorialAppInputStream (deviceID string, roomNo int, temperature double);\n  -- Table\n  CREATE TABLE GLOBAL tutorialAppOutputTable (id string, temperature double);\n  @info(name=\'Query\')\n  INSERT INTO tutorialAppOutputTable\n  SELECT concatFn(roomNo,\'-\',deviceID) as id, temperature\n  FROM tutorialAppInputStream;\n', 'regions': ['play-us-west'], 'isActive': False}]}

stream worker Activated:  {'error': False, 'code': 200, 'streamApps': [{'name': 'stream_worker_tutorial', 'definition': '\n  @App:name(\'stream_worker_tutorial\')\n  @App:qlVersion("2")\n  CREATE FUNCTION concatFn[javascript] return string {\n      var str1 = data[0];\n      var str2 = data[1];\n      var str3 = data[2];\n      var response = str1 + str2 + str3;\n      return response;\n  };\n  -- Stream\n  CREATE STREAM tutorialAppInputStream (deviceID string, roomNo int, temperature double);\n  -- Table\n  CREATE TABLE GLOBAL tutorialAppOutputTable (id string, temperature double);\n  @info(name=\'Query\')\n  INSERT INTO tutorialAppOutputTable\n  SELECT concatFn(roomNo,\'-\',deviceID) as id, temperature\n  FROM tutorialAppInputStream;\n', 'regions': ['play-us-west'], 'isActive': True}]}
Message published successfully
Message published successfully

stream worker Results:  {'result': [{'_id': 'tutorialAppOutputTable/1983609465', '_key': '1983609465', '_rev': '_fWH5i2y--_', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1983609467', '_key': '1983609467', '_rev': '_fWH5i4a--_', 'id': '201-AD11', 'temperature': 47}, {'_id': 'tutorialAppOutputTable/1984084894', '_key': '1984084894', '_rev': '_fWIMMmq--_', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1984084896', '_key': '1984084896', '_rev': '_fWIMMmu--_', 'id': '201-AD11', 'temperature': 47}, {'_id': 'tutorialAppOutputTable/1984095390', '_key': '1984095390', '_rev': '_fWIMiHq--_', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1984101014', '_key': '1984101014', '_rev': '_fWIMr5K--_', 'id': '201-AD11', 'temperature': 47}, {'_id': 'tutorialAppOutputTable/1984155228', '_key': '1984155228', '_rev': '_fWIO6Ze--_', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1984155234', '_key': '1984155234', '_rev': '_fWIO6Zm--_', 'id': '201-AD11', 'temperature': 47}, {'_id': 'tutorialAppOutputTable/1984155236', '_key': '1984155236', '_rev': '_fWIO6Zm--C', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1984155239', '_key': '1984155239', '_rev': '_fWIO6Zq--B', 'id': '201-AD11', 'temperature': 47}, {'_id': 'tutorialAppOutputTable/1984164734', '_key': '1984164734', '_rev': '_fWIPP0m--_', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1984170401', '_key': '1984170401', '_rev': '_fWIPZk2--_', 'id': '201-AD11', 'temperature': 47}, {'_id': 'tutorialAppOutputTable/1984220920', '_key': '1984220920', '_rev': '_fWIRUnm--_', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1984220923', '_key': '1984220923', '_rev': '_fWIRUnu--_', 'id': '201-AD11', 'temperature': 47}, {'_id': 'tutorialAppOutputTable/1984220925', '_key': '1984220925', '_rev': '_fWIRUny--_', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1984220927', '_key': '1984220927', '_rev': '_fWIRUny--C', 'id': '201-AD11', 'temperature': 47}, {'_id': 'tutorialAppOutputTable/1984220929', '_key': '1984220929', '_rev': '_fWIRUn2--_', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1984220931', '_key': '1984220931', '_rev': '_fWIRUn2--C', 'id': '201-AD11', 'temperature': 47}, {'_id': 'tutorialAppOutputTable/1984221812', '_key': '1984221812', '_rev': '_fWIRa6a--_', 'id': '200-AD11', 'temperature': 18}, {'_id': 'tutorialAppOutputTable/1984221878', '_key': '1984221878', '_rev': '_fWIRa8i--_', 'id': '201-AD11', 'temperature': 47}], 'hasMore': False, 'cached': False, 'extra': {'stats': {'writesExecuted': 0, 'writesIgnored': 0, 'scannedFull': 20, 'scannedIndex': 0, 'filtered': 0, 'httpRequests': 0, 'executionTime': 0.00016641616821289062, 'peakMemoryUsage': 24008}, 'warnings': []}, 'error': False, 'code': 201}

stream worker Deleted:  {'error': False, 'code': 200, 'message': 'Stream worker `stream_worker_tutorial` deleted successfully.'}
```

**Pull request merged**